// @progress/kendo-react-layout/dist/npm/timeline/TimelineCard.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineCard = void 0;
var React = require("react");
var PropTypes = require("prop-types");
var Card_1 = require("../card/Card");
var CardHeader_1 = require("../card/CardHeader");
var CardTitle_1 = require("../card/CardTitle");
var CardSubtitle_1 = require("../card/CardSubtitle");
var CardBody_1 = require("../card/CardBody");
var CardImage_1 = require("../card/CardImage");
var CardActions_1 = require("../card/CardActions");
var kendo_react_buttons_1 = require("@progress/kendo-react-buttons");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var kendo_react_animation_1 = require("@progress/kendo-react-animation");
var kendo_svg_icons_1 = require("@progress/kendo-svg-icons");
var TimelineCard = function (props) {
    var animationDuration = 400;
    var _a = props.eventData, title = _a.title, subtitle = _a.subtitle, actions = _a.actions, images = _a.images, description = _a.description;
    var _b = React.useState(true), collapse = _b[0], setCollapse = _b[1];
    var _c = React.useState(true), collapsed = _c[0], setCollapsed = _c[1];
    var toggleCard = function (event) {
        setCollapse(!collapse);
        if (props.onChange) {
            var toggleEvent = {
                syntheticEvent: event,
                nativeEvent: event.nativeEvent,
                eventData: props.eventData
            };
            props.onChange.call(undefined, toggleEvent);
        }
    };
    var actionClick = function (event) {
        if (props.onActionClick) {
            var actionClickEvent = {
                syntheticEvent: event,
                nativeEvent: event.nativeEvent,
                eventData: props.eventData
            };
            props.onActionClick.call(undefined, actionClickEvent);
        }
    };
    var cardBody = function () { return (React.createElement(CardBody_1.CardBody, null,
        React.createElement("div", { className: "k-card-description" },
            React.createElement("p", null, description),
            images && images.map(function (image, index) { return (React.createElement(CardImage_1.CardImage, { key: index, src: image.src })); })),
        React.createElement(CardActions_1.CardActions, null, actions && actions.map(function (action, index) { return (React.createElement("a", { key: index, href: action.url, className: "k-button k-button-md k-rounded-md k-button-flat k-button-flat-primary", onClick: actionClick }, action.text)); })))); };
    return (React.createElement("div", { className: (0, kendo_react_common_1.classNames)('k-timeline-card', { 'k-collapsed': props.collapsible && collapsed }) },
        React.createElement(Card_1.Card, { className: "k-card-with-callout" },
            React.createElement("span", { className: (0, kendo_react_common_1.classNames)('k-timeline-card-callout', 'k-card-callout', { 'k-callout-e': props.alternated }, { 'k-callout-w': !props.alternated }) }),
            React.createElement("div", { className: "k-card-inner" },
                React.createElement(CardHeader_1.CardHeader, null,
                    React.createElement(CardTitle_1.CardTitle, null,
                        React.createElement("span", { className: "k-event-title" }, title),
                        props.collapsible && React.createElement(kendo_react_buttons_1.Button, { fillMode: 'flat', icon: 'chevron-right', svgIcon: kendo_svg_icons_1.chevronRightIcon, onClick: function (event) { return toggleCard(event); }, className: 'k-event-collapse' })),
                    subtitle && React.createElement(CardSubtitle_1.CardSubtitle, null, subtitle)),
                props.collapsible ?
                    React.createElement(kendo_react_animation_1.Reveal, { transitionEnterDuration: animationDuration, transitionExitDuration: animationDuration, onBeforeEnter: function () { return setCollapsed(false); }, onAfterExited: function () { return setCollapsed(true); } }, !collapse ? cardBody() : null)
                    : cardBody()))));
};
exports.TimelineCard = TimelineCard;
exports.TimelineCard.propTypes = {
    onChange: PropTypes.func,
    onActionClick: PropTypes.func
};
